exonum-java-binding/core/rust/src/storage/db.rs

```diff
@@ -28,104 +28,120 @@ pub(crate) type Value = Vec<u8>;
 ///    in the case when the java side creates and owns it;
 /// - it just holds a reference, when one is provided from the rust side.
 ///
-/// For storage API we need a reference, so we create it to the owned part. But since there is no
-/// way in Rust to make a `View` value not movable. Furthermore, it have to be moved from the stack
-/// to the heap in order to be converted into `Handle` for the java side. So a `Fork` value
-/// should be placed in the heap to prevent its movement after creating a reference to it.
-
-pub(crate) struct View {
-    /// Unsafe reference to the owned View (owned either by Rust or by Java)
-    reference: ViewRef,
-    /// Allows to drop owned view if needed
-    owned: Option<ViewOwned>,
+/// As `View` does not have a lifetime, nothing protects us from dereferencing a freed reference,
```

Comment:
I don't get what you mean here by 'dereferencing a freed reference' and how it is connected with View instantiation.

Reply:
The View does not have a lifetime, so we use `unsafe` to prolong the lifetime of the reference it was constructed with to `'static`. If this original reference is destroyed (leaves the scope), our prolonged reference stored inside View is no longer valid and will lead to SIGINT if we use it. So we must carefully review all the places where View is constructed from references to make sure View never outlives the original reference. Does it make sense now?

I agree, `freed` cannot be used in case of references here, I will rephrase.

Reply:
Yes, thank you, that's clearer